### PR TITLE
HFB add beam container

### PR DESCRIPTION
**Pull request by rikvl**
Make a base class `HFBBeamContainer` for all HFB containers that have
that have a `beam` axis. This also has `beam_ew` and `beam_ns`

I think this code could be looked at already, ~~but merging may need to wait for PR #210 to be merged~~

Note to self: ~~after PR #210 is handled, but~~ before merging this PR, check if any changes still need to be added in `hfb/analysis.py` from commit 21a267c8 that could otherwise get lost

### CODE REVIEW

File: ch_pipeline/hfb/containers.py [hunk 1]
```diff
@@ -51,7 +52,32 @@ def nsample(self) -> memh5.MemDataset:
...
+        """The beam indices associated with each entry of the beam axis."""
+        return self.index_map["beam"]
+
+    @cached_property
+    def beam_ew(self):
+        """The unique EW-beam indices (i.e., from 0 to 3) in the beam axis."""
+        return np.unique(self.beam // 256)
+
+    @cached_property
+    def beam_ns(self):
```
Comment (jrs65) [inline ch_pipeline/hfb/containers.py:75]: Maybe add type annotations to these.

**Review by jrs65 — DISMISSED** [review]
Looks good, just a few comments about annotations and docstrings. Just merge when you've made the changes, no need for re-review.

File: ch_pipeline/hfb/containers.py (hunk 1 above)
Comment (rikvl) [reply ch_pipeline/hfb/containers.py]: Done.

File: ch_pipeline/hfb/containers.py [hunk 2]
```diff
@@ -341,15 +363,43 @@ class HFBHighResSpectrum(HFBHighResContainer):
...
+
+    @property
+    def beam_ns(self):
+        """The (unique) NS beam indices (i.e., from 0 to 256) of the beam_ns axis."""
+        return self.index_map["beam_ns"]
 
     @property
     def el(self):
-        """The elevation in degrees associated with each sample of the elevation axis."""
+        """The el = sin(zenith angle) associated with each sample of the el axis.
```
[the remark this answers is not in the document]
Comment (rikvl) [reply ch_pipeline/hfb/containers.py:390]: Done.

File: ch_pipeline/hfb/containers.py [hunk 3]
```diff
@@ -51,7 +52,32 @@ def nsample(self) -> memh5.MemDataset:
...
+        return self.index_map["beam"]
+
+    @cached_property
+    def beam_ew(self) -> np.ndarray:
+        """The unique EW-beam indices (i.e., from 0 to 3) in the beam axis."""
+        return np.unique(self.beam // 256)
+
+    @cached_property
+    def beam_ns(self) -> np.ndarray:
+        """The unique NS-beam indices (i.e., from 0 to 256) in the beam axis."""
```
Comment (arashcosmology) [inline ch_pipeline/hfb/containers.py:76]: NS-beam indices run from 0 to 255 (and not from 0 to 256).

Comment (rikvl) [reply ch_pipeline/hfb/containers.py]: Fixed.

**Review by jrs65 — APPROVED** [review]

**Pull request merged**